External/FEXCore/Source/Interface/Core/OpcodeDispatcher.cpp

```diff
@@ -2556,23 +2556,21 @@ void OpDispatchBuilder::IMULOp(OpcodeArgs) {
   if (Size == 1) {
     // Result is stored in AX
     _StoreContext(GPRClass, 2, offsetof(FEXCore::Core::CPUState, gregs[FEXCore::X86State::REG_RAX]), Result);
-    ResultHigh = _Bfe(8, 8, Result);
-    ResultHigh = _Sext(Size * 8, ResultHigh);
+    ResultHigh = _Sbfe(8, 8, Result);
   }
   else if (Size == 2) {
     // 16bits stored in AX
     // 16bits stored in DX
     _StoreContext(GPRClass, Size, offsetof(FEXCore::Core::CPUState, gregs[FEXCore::X86State::REG_RAX]), Result);
-    ResultHigh = _Bfe(16, 16, Result);
-    ResultHigh = _Sext(Size * 8, ResultHigh);
+    ResultHigh = _Sbfe(16, 16, Result);
     _StoreContext(GPRClass, Size, offsetof(FEXCore::Core::CPUState, gregs[FEXCore::X86State::REG_RDX]), ResultHigh);
   }
   else if (Size == 4) {
     // 32bits stored in EAX
     // 32bits stored in EDX
     // Make sure they get Zext correctly
-    OrderedNode *ResultLow = _Bfe(4, 32, 0, Result);
-    ResultHigh = _Bfe(4, 32, 32, Result);
+    OrderedNode *ResultLow = _Sbfe(32, 0, Result);
```

Comment:
Shouldn't these remain `_Bfe`s here?

Reply:
(I went over the manual and it doesn't say anything about SEXT on imul, so don't the typical ZEXT rules apply?)

Reply:
Oops, you're right.
I was abusing the sext for the flags calculation in the 8bit and 16bit flags calculation and accidentally extended in to the 32bit side

Reply:
Should probably extend the unit test again just to ensure proper zext mechanics here.